pkg/frontend/frontend.go

```diff
@@ -139,7 +142,18 @@ func NewFrontend(cfg Config, limits Limits, log log.Logger, reg prometheus.Regis
 		return nil, err
 	}
 
+	settingsStore, err := settings.NewMemoryStore()
```

Comment:
This is actually a query frontend, that is there only for speeding up and applying fair scheduling in a multi-tenant setup.

Happy that we start like this but we might want to move this in another component.

Reply:
I think how I have it configured as of https://github.com/grafana/pyroscope/pull/2711/commits/600bdb6301754060ead1667bb2128e7dd6d93ce8, it will only run in single binary mode.

Longer term, do we want to deploy a separate component to handle settings? Or do add the settings module to queriers? It might be more effort than its worth to have the settings module running as part of a ring, especially for something that's so small right now.

Reply:
That sounds great !